File: SingleNumber/SingleNum.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        # if len(nums) == 1: return int(''.join(map(str, nums))) -> Эта строка необязательна, она делает из списка число
        x = 0
        for i in nums:
            # ^ -> Проводит побитовую операцию xor (исключающее или) на двух значениях. x = x ^ i
            x ^= i
        return x


class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        return reduce(lambda total, el: total ^ el, nums)


class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        for i in nums:
            if nums.count(i) == 1:
                return i

File: SingleNumber/SingleNum.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function (nums) {
  if (nums.length === 1) {
    return nums[0];
  }
  let result = 0;
  nums.forEach((element) => {
    // ^ -> Проводит побитовую операцию xor (исключающее или) на двух значениях. x = x ^ i
    result = result ^ element;
  });
  return result;
};


// ! Solution 2
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function (nums) {
  let i = 0;
  nums.forEach((e) => {
    if (nums.filter((n) => n === e).length === 1) i = e;
  });
  return i;
};
